woodwork/type_sys/inference_functions.py

```diff
@@ -4,27 +4,15 @@
 import woodwork as ww
 from woodwork.type_sys.utils import _is_categorical_series, col_is_datetime
 
-INFERENCE_SAMPLE_SIZE = 10000
 
-
-def get_inference_sample(series: pd.Series) -> pd.Series:
-    f"""
-    Return a sample of ``series`` for use during type inference.  If the length
-    of ``series`` is less than ``{INFERENCE_SAMPLE_SIZE}``, use the series
-    length as the sample size.
-    """
-    return series.sample(n=min(INFERENCE_SAMPLE_SIZE, len(series)))
-
-
-def categorical_func(series):
+def categorical_func(series: pd.Series) -> bool:
```

Comment:
Maybe this should be a bigger discussion, but we haven't been doing type annotations so far, so not sure we should include this here?

Reply:
I've just been adding them as I go along.  I'm a pretty big advocate of type annotations.  Unless we're running mypy on the codebase, adding annotations basically has no effect.  But it can still serve as useful information for developers in understanding how a function behaves.  Are you concerned that the annotations will become stale somehow if we're not using them for type checking?

I should mention that I did bring the topic up with a few other people at Inno Labs and it seemed like the general sentiment was positive for this type of change.

Reply:
Don't get me wrong, I'm not anti-annotations by any means, but it's not a practice we have adopted broadly so far. Was just thinking that if this a direction we collectively want to head, we should have that discussion so the main team is aware of how we want to do things going forward, and that seems to be happening now...

Reply:
Yep, valid point.

Reply:
I just took it out for now.